source/_posts/2015-10-11-a-configuration-management-rosetta-stone-configuring-sensu-with-puppet.markdown

```diff
@@ -0,0 +1,352 @@
+other RabbitMQ, Redis, or any other module. To me this is expected, in the
+Puppet world it would be the job of a `profile` to combine the Sensu module
+with RabbitMQ and other things. For the most part this integration is left as
+an exercise to the reader.
+
+The Sensu Puppet module also doesn't manage Uchiwa. That requires a
+[different puppet module](https://github.com/Yelp/puppet-uchiwa). Again to me
+this is a good thing, I hate it when tools try to do too much.
+
+The actual codebase is actively maintained and reasonably active, with a few
+releases per year. The Puppet Forge has almost perfect scores for it's quality
+ratings. The code has excellent unit test and acceptance test coverage. As far
+as puppet modules go, the Sensu puppet module is a great example of a
+well-maintained piece of code.
+
+One downside the "completeness" of the module is that sometimes new features of
+Sensu are released, and the puppet-module will lag. The configuration inputs to
+the puppet module are well-typed, and not just free-form hashes. This gives a
+lot of guard-rails and helps ensure config files are correct before they hit
+the disk, but it means that some features are not usable until the Puppet
+module can account for them.
+
+Although the code worked, there was a [significant bug](https://github.com/sensu/sensu-puppet/issues/336) that prevented the
+module from ever converging. This was annoying but allowed me to test the code.
+This bug looks to be fixed in master.
+
+### Chef
+
+#### Chef in General
+
+Chef is not as old as Puppet, but is certainly a mature product. Chef is "just
+ruby" when it comes to it's configuration language. The upside to this is that
+Ruby developers can theortically dive in and hack on stuff. The downside to
+this is that being "just ruby", "leaves a lot of rope to hang yourself".
+
+One nice feature provided by the Chef-company is their hosted-chef solution, which
+allows people to get started without hosting a Chef-server.
+
+The Chef toolset also comes with the `knife` command, which is a great command
+line tool for interacting with the Chef-server. It also is a parallel-ssh tool,
+manipulates chef cookbooks, and can also launch ec2 (and other) instances. (did
+they take the kitchen-sink metaphor too far?)
+
+The [Chef Supermarket](https://supermarket.chef.io/) serves as the public
+registry for Chef cookbooks. There are not too many quality indicators to see,
+to help find which cookbooks are any good. The best metric I could see is just
+sorting by "followers". This is made up by the fact the fact that there is over
+a hundred [officially supported
+cookbooks](https://supermarket.chef.io/users/chef). 
+
+Probably the most difficult aspect of Chef for me to understand was how
+attributes interact.  This confusion probably most obvious when you look at
+Chef's [15 levels of attribute
+precedence](https://docs.chef.io/attributes.html#attribute-precedence). Seems
```

Comment:
This is totally a style thing, but I usually don't omit the initial noun here.  I would say, "It seems to me", but I can also totally understand if you're trying to strike a colloquial tone.